cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/CapacityGoal.java

```diff
@@ -52,32 +54,46 @@ public CapacityGoal() {
 
   protected abstract Resource resource();
 
+  /**
+   * @deprecated Please use {@link this#actionAcceptance(BalancingAction, ClusterModel)} instead.
+   */
+  @Override
+  public boolean isActionAcceptable(BalancingAction action, ClusterModel clusterModel) {
+    return actionAcceptance(action, clusterModel) == ACCEPT;
+  }
+
   /**
    * Check whether the given action is acceptable by this goal. An action is acceptable by a goal if it satisfies
    * requirements of the goal. Requirements(hard goal): Capacity.
    *
    * ## Leadership Movement: impacts only (1) network outbound and (2) CPU resources (See
-   * {@link DiskCapacityGoal#isActionAcceptable(BalancingAction, ClusterModel)} and
-   * {@link NetworkInboundCapacityGoal#isActionAcceptable(BalancingAction, ClusterModel)}).
+   * {@link DiskCapacityGoal#actionAcceptance(BalancingAction, ClusterModel)} and
+   * {@link NetworkInboundCapacityGoal#actionAcceptance(BalancingAction, ClusterModel)}).
    *   (1) Check if leadership NW_OUT movement is acceptable: NW_OUT movement carries all of leader's NW_OUT load.
    *   (2) Check if leadership CPU movement is acceptable: In reality, CPU movement carries only a fraction of
    * leader's CPU load.
    * To optimize CC performance, we avoid calculation of the expected leadership CPU utilization, and assume that
-   * if (action.balancingAction().equals(ActionType.LEADERSHIP_MOVEMENT) && resource().equals(Resource.CPU)),
+   * if (action.balancingAction() == ActionType.LEADERSHIP_MOVEMENT && resource() == Resource.CPU),
    * then the expected leadership CPU utilization would be the full CPU utilization of the leader.
    * <p>
    * ## Replica Movement: impacts any resource.
+   * ## Replica Swap: impacts any resource.
    *
    * @param action Action to be checked for acceptance.
    * @param clusterModel The state of the cluster.
-   * @return True if the action is acceptable by this goal, false otherwise.
+   * @return {@link ActionAcceptance#ACCEPT} if the action is acceptable by this goal,
+   * {@link ActionAcceptance#REPLICA_REJECT} otherwise.
    */
   @Override
-  public boolean isActionAcceptable(BalancingAction action, ClusterModel clusterModel) {
+  public ActionAcceptance actionAcceptance(BalancingAction action, ClusterModel clusterModel) {
     Replica sourceReplica = clusterModel.broker(action.sourceBrokerId()).replica(action.topicPartition());
     Broker destinationBroker = clusterModel.broker(action.destinationBrokerId());
 
-    return isMovementAcceptableForCapacity(resource(), sourceReplica, destinationBroker);
+    if (action.balancingAction() == ActionType.REPLICA_SWAP) {
```

Comment:
Can we use switch statement here? It would be less error prone to make sure are not missing any case.